src/query/util/opentracing/context.go

```diff
@@ -0,0 +1,80 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package opentracingutil provides more utilities for dealing with opentracing and context.Context's.
+package opentracingutil
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+// alias GlobalTracer() so we can mock out the tracer without impacting tests outside the package
+var getGlobalTracer = opentracing.GlobalTracer
+
+// SpanFromContextOrRoot is the same as opentracing.SpanFromContext, but instead of returning nil,
+// it returns a NoopTracer span if ctx doesn't already have an associated span.
+// Use this over opentracing.StartSpanFromContext if you need access to the
+// current span, (e.g. if you don't want to start a child span).
+//
+// NB: if there is no span in the context, the span returned by this function
+// is a noop, and won't be attached to the context; if you
+// want a proper span, either start one and pass it in, or start one
+// in your function.
+func SpanFromContextOrNoop(ctx context.Context) opentracing.Span {
+	sp := opentracing.SpanFromContext(ctx)
+	if sp != nil {
+		return sp
+	}
+
+	return opentracing.NoopTracer{}.StartSpan("")
+}
+
+// StartSpanFromContext is the same as opentracing.StartSpanFromContext, but instead of always using the global tracer,
+// it attempts to use the parent span's tracer if it's available. This behavior is (arguably) more flexible--it allows
+// a locally set tracer to be used when needed (as in tests)--while being equivalent to the original in most contexts.
+// See https://github.com/opentracing/opentracing-go/issues/149 for more discussion.
+func StartSpanFromContext(ctx context.Context, operationName string, opts ...opentracing.StartSpanOption) (opentracing.Span, context.Context) {
+	var span opentracing.Span
+	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
+		opts = append(opts, opentracing.ChildOf(parentSpan.Context()))
+		tracer := parentSpan.Tracer()
+		span = tracer.StartSpan(operationName, opts...)
+	} else {
+		tracer := getGlobalTracer()
+		span = tracer.StartSpan(operationName, opts...)
```

Comment:
Pull out `span = tracer.StartSpan(operationName, opts...)` since its in both conditional statements?